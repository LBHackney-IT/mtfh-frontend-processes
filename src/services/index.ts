export { default as locale } from "./locale";
export { default as menu } from "./menu";
export * as processes from "./processes";
export * from "./config";
