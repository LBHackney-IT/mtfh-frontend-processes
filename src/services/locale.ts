import reactHtmlParser from "react-html-parser";

import { AssetAddress } from "@mtfh/common/lib/api/asset/v1";

const locale = {
  backButton: "Back",
  bookAppointment: "Book Appointment",
  next: "Next",
  cancel: "Cancel",
  cancelProcess: "Cancel process",
  change: "Change",
  views: {
    soleToJoint: {
      title: "Sole tenant requests a joint tenure",
      steps: {
        selectTenant: "Select person",
        checkEligibility: "Eligibility Checks",
        supportingDocuments: "Supporting documents",
        finish: "Finish",
        breachOfTenancy: "Breach of tenure check",
        requestDocuments: "Request Documents",
        reviewDocuments: "Review Documents",
        submitCase: "Submit case",
      },
      actions: {
        reassignCase: "Reassign case",
        cancelProcess: "Cancel process",
        caseActivityHistory: "Case activity history",
      },
    },
    selectTenants: {
      selectTenantHint:
        "This person will be asked for proof of relationship e.g. marriage or civil partnership certificate.",
      selectTenantLabel: "Who do you want to add as a joint tenant?",
      addToTenureText: "If the person you want to add is not listed you must first",
      addToTenureLink: "add them to the tenure",
      noHouseholdMembersOver18:
        "There are no household members over the age of 18 associated to this.",
    },
    processesMenu: {
      selectProcessDescription: () =>
        reactHtmlParser(
          "<p>Start a process by selecting a process category below. <br /> Where a process is not yet supported in <strong>Manage My Home</strong> these links will open the appropriate interim Google form.</p>",
        ),
    },
    checkEligibility: {
      autoCheckIntro:
        "Manage My Home has used tenure, person and property information to automatically check eligibility for this application.",
      passedChecks: "Passed automatic eligibilty checks",
      failedChecks: "Not eligible for a sole to joint tenure",
    },
    breachOfTenancy: {
      autoCheckIntro:
        "Manage My Home has used tenure, person and property information to automatically check eligibility for this application.",
      passedChecks: "Eligibility checks passed",
      failedChecks: "Not eligible for a sole to joint tenure",
    },
    reviewDocuments: {
      passedChecks: "Eligibility checks passed",
      documentsRequested: "Documents requested",
      checkSupportingDocumentsAppointment:
        "I need to make an appointment to check supporting documents",
      appointmentScheduled: "Office appointment scheduled",
      seenPhotographicId:
        "I confirm that an outcome letter has been sent to the resident",
      seenSecondId:
        "I confirm I have seen a second form of ID (does not have to be photographic)",
      isNotInImmigrationControl:
        "I confirm that the prospective tenant is not subject to immigration control under the Asylum And Immigration Act 1996",
      seenProofOfRelationship:
        "I confirm I have seen proof of relationship to the existing tenant",
      incomingTenantLivingInProperty:
        "I confirm I have seen 3 separate documents proving the proposed tenant has been living at the property for a minimum of 12 months",
    },
  },
  errors: {
    unableToFetchRecord: "There was a problem retrieving the record",
    unableToFetchRecordDescription:
      "Please try again. If the issue persists, please contact support.",
    unableToFindProcess: "Invalid Process",
    unableToFindProcessDescription: "We have been unable to find a process of this type.",
    unableToFindState: "Invalid State",
    unableToFindStateDescription: "We have been unable to process a state of this type.",
  },
  title: "Processes",
  loadingText: "Submitting...",
  components: {
    startProcess: {
      buttonLabel: "Start process",
      thirdPartyHeading: "Sharing information with third parties",
    },
    entitySummary: {
      tenurePaymentRef: "Tenure payment ref",
      address: (assetAddress: AssetAddress): string => {
        const {
          postPreamble,
          addressLine1,
          addressLine2,
          addressLine3,
          addressLine4,
          postCode,
        } = assetAddress;

        return [
          postPreamble,
          addressLine1,
          addressLine2,
          addressLine3,

          addressLine4,
          postCode,
        ]
          .filter((addressLine) => !!addressLine)
          .join(" ");
      },
    },
  },
};

export default locale;
