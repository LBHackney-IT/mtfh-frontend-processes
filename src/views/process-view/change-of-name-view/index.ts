export * from "./change-of-name-view";
export * from "./states";
