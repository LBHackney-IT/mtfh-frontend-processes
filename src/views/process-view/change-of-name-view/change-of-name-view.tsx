import React from "react";
import { Link as RouterLink } from "react-router-dom";

import { locale, processes } from "../../../services";
import { ProcessSideBarProps } from "../../../types";
import { RequestDocumentsView } from "./states/request-documents-view";

import { usePerson } from "@mtfh/common/lib/api/person/v1";
import { Process } from "@mtfh/common/lib/api/process/v1";
import {
  Button,
  Center,
  ErrorSummary,
  Spinner,
  Step,
  Stepper,
} from "@mtfh/common/lib/components";

import "./styles.scss";

const processConfig = processes.changeofname;

const { states } = processConfig;
const { enterNewName, nameSubmitted } = states;

const components = {
  [enterNewName.state]: RequestDocumentsView,
  [nameSubmitted.state]: RequestDocumentsView,
};

const { views } = locale;
const { changeofname } = views;

const getActiveStep = () => {
  return 0;
};

export const ChangeOfNameSideBar = (props: ProcessSideBarProps) => {
  const {
    process: { processId, processName },
    setCloseProcessDialogOpen,
    setCancel,
  } = props;

  const activeStep = getActiveStep();
  const steps: JSX.Element[] = [
    <Step key="step-breach-of-tenancy">{changeofname.steps.tenantsNewName}</Step>,
    <Step key="step-request-documents">{changeofname.steps.requestDocuments}</Step>,
    <Step key="step-review-documents">{changeofname.steps.reviewDocuments}</Step>,
    <Step key="step-submit-case">{changeofname.steps.submitCase}</Step>,
    <Step key="step-finish">{changeofname.steps.finish}</Step>,
  ];
  const startIndex = 0;

  return (
    <>
      <Stepper
        data-testid="mtfh-stepper-change-of-name"
        activeStep={activeStep}
        startIndex={startIndex}
      >
        {steps}
      </Stepper>
      <Button variant="secondary">{changeofname.actions.reassignCase}</Button>
      <Button
        variant="secondary"
        onClick={() => {
          setCancel(true);
          setCloseProcessDialogOpen(true);
        }}
      >
        {changeofname.actions.cancelProcess}
      </Button>
      <Button
        variant="secondary"
        as={RouterLink}
        to={`/activities/process/${processName}/${processId}`}
      >
        {changeofname.actions.caseActivityHistory}
      </Button>
    </>
  );
};

const getComponent = (process) => {
  const {
    currentState: { state },
  } = process;

  return components[state];
};

export const ChangeOfNameView = ({
  process,
  mutate,
  optional,
}: {
  process: Process;
  mutate: any;
  optional: any;
}): JSX.Element => {
  const { closeProcessReasonFinal, submitted, setSubmitted, closeCase, setCloseCase } =
    optional;
  const { error, data: person } = usePerson(process.targetId);

  if (error) {
    return (
      <ErrorSummary
        id="entity-summary"
        title={locale.errors.unableToFetchRecord}
        description={locale.errors.unableToFetchRecordDescription}
      />
    );
  }

  if (!person) {
    return (
      <Center>
        <Spinner />
      </Center>
    );
  }

  const Component = getComponent(process);

  if (!Component) {
    return (
      <ErrorSummary
        id="change-of-name-view"
        title={locale.errors.unableToFindState}
        description={locale.errors.unableToFindStateDescription}
      />
    );
  }

  return (
    <Component
      processConfig={processConfig}
      process={process}
      mutate={mutate}
      optional={{
        person,
        submitted,
        setSubmitted,
        closeCase,
        setCloseCase,
        closeProcessReasonFinal,
      }}
    />
  );
};
