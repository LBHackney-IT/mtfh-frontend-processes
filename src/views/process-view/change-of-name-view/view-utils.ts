import { locale, processes } from "../../../services";

const { reviewDocuments } = locale.views;

const { states } = processes.changeofname;

const { interviewScheduled, interviewRescheduled, hoApprovalFailed, hoApprovalPassed } =
  states;

export const changeOfNameDocuments = [
  reviewDocuments.marriageCertificate,
  reviewDocuments.civilPartnershipCertificate,
  reviewDocuments.decreeAbsolute,
  reviewDocuments.finalOrder,
  reviewDocuments.deedPollDocument,
  reviewDocuments.statuoryDeclaration,
];

export const cancelButtonStates = [
  interviewScheduled.state,
  interviewRescheduled.state,
  hoApprovalFailed.state,
  hoApprovalPassed.state,
];
