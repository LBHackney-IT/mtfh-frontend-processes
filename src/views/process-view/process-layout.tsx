import React, { useState } from "react";
import { Link as RouterLink, useParams } from "react-router-dom";

import {
  ChangeOfNameSideBar,
  ChangeOfNameView,
  SoleToJointSideBar,
  SoleToJointView,
} from "..";
import { CloseProcessDialog } from "../../components";
import { locale, processes } from "../../services";
import { CommentsView } from "./shared/comments-view/comments-view";

import { useProcess } from "@mtfh/common/lib/api/process/v1";
import {
  Center,
  ErrorSummary,
  Heading,
  Layout,
  Link,
  Spinner,
} from "@mtfh/common/lib/components";

const components = {
  [processes.soletojoint.processName]: SoleToJointView,
  [processes.changeofname.processName]: ChangeOfNameView,
};

const sidebars = {
  [processes.soletojoint.processName]: SoleToJointSideBar,
  [processes.changeofname.processName]: ChangeOfNameSideBar,
};

const isSameState = (firstState, secondState) => {
  return firstState.state === secondState.state;
};

export const ProcessLayout = (): JSX.Element => {
  const { processId, processName } =
    useParams<{ processId: string; processName: string }>();

  const [isCloseProcessDialogOpen, setCloseProcessDialogOpen] = useState<boolean>(false);
  const [closeProcessReason, setCloseProcessReason] = useState<string>();
  const [isCancel, setCancel] = useState<boolean>(false);
  const [submitted, setSubmitted] = useState<boolean>(false);
  const [closeCase, setCloseCase] = useState<boolean>(false);

  const {
    data: process,
    error,
    mutate,
  } = useProcess({
    id: processId,
    processName,
  });

  if (error) {
    return (
      <ErrorSummary
        id="sole-to-joint-view"
        title={locale.errors.unableToFetchRecord}
        description={locale.errors.unableToFetchRecordDescription}
      />
    );
  }

  if (!process) {
    return (
      <Center>
        <Spinner />
      </Center>
    );
  }

  const Component = components[processName];
  const Sidebar = sidebars[processName];

  if (!Component) {
    return (
      <ErrorSummary
        id="process-layout-error"
        title={locale.errors.unableToFindState}
        description={locale.errors.unableToFindStateDescription}
      />
    );
  }

  const processConfig = processName && processes[processName];
  const { states } = processConfig;
  const { processClosed } = states;

  let closeProcessReasonFinal = closeProcessReason;
  if (!closeProcessReasonFinal && isSameState(process.currentState, processClosed)) {
    closeProcessReasonFinal = process.currentState.processData.formData.Reason;
  }

  return (
    <Layout
      data-testid="process-layout"
      sidePosition="right"
      backLink={
        <Link
          as={RouterLink}
          to={`/processes/${processName}/start/${process.targetType}/${process.targetId}`}
          variant="back-link"
        >
          {locale.backButton}
        </Link>
      }
      side={
        <Sidebar
          process={process}
          submitted={submitted}
          closeCase={closeCase}
          setCloseProcessDialogOpen={setCloseProcessDialogOpen}
          setCancel={setCancel}
        />
      }
    >
      <Heading variant="h1">{processConfig.title}</Heading>
      <Component
        process={process}
        mutate={mutate}
        optional={{
          submitted,
          setSubmitted,
          closeCase,
          setCloseCase,
          setCancel,
          setCloseProcessDialogOpen,
          closeProcessReasonFinal,
        }}
      />

      <CloseProcessDialog
        isCloseProcessDialogOpen={isCloseProcessDialogOpen}
        setCloseProcessDialogOpen={setCloseProcessDialogOpen}
        setCloseProcessReason={setCloseProcessReason}
        mutate={mutate}
        isCancel={isCancel}
      />

      <hr className="divider" />
      <CommentsView targetType="process" targetId={processId} mutate={mutate} />
    </Layout>
  );
};
