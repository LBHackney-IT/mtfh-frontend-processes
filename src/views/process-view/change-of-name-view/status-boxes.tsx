import React from "react";
import { Link as RouterLink } from "react-router-dom";

import { locale } from "../../../services";
import { isPreviousState } from "../../../utils/processUtil";
import { DesBox } from "../shared/process-components";
import { TenureInvestigationRecommendationBox } from "../sole-to-joint-view/states/shared";
import { tenureInvestigationResultStates } from "./view-utils";

import { Link, StatusBox, Text } from "@mtfh/common/lib/components";

export const StatusBoxes = ({ process, processConfig }) => {
  const { states } = processConfig;
  const {
    documentsRequestedDes,
    documentChecksPassed,
    hoApprovalFailed,
    hoApprovalPassed,
    nameUpdated,
  } = states;

  const hoApprovalPassedState =
    process.currentState.state === hoApprovalPassed.state
      ? process.currentState
      : process.previousStates.find(
          (previous) => previous.state === hoApprovalPassed.state,
        );

  const hoApprovalFailedState =
    process.currentState.state === hoApprovalFailed.state
      ? process.currentState
      : process.previousStates.find(
          (previous) => previous.state === hoApprovalFailed.state,
        );

  return (
    <>
      {documentChecksPassed.state === process.currentState.state ||
      isPreviousState(documentChecksPassed.state, process) ? (
        <DesBox
          title={locale.views.submitCase.supportingDocumentsApproved}
          description={locale.views.submitCase.viewDocumentsOnDes}
        />
      ) : (
        (documentsRequestedDes.state === process.currentState.state ||
          isPreviousState(documentsRequestedDes.state, process)) && (
          <DesBox title={locale.views.reviewDocuments.documentsRequested} />
        )
      )}

      {(tenureInvestigationResultStates.includes(process.currentState.state) ||
        tenureInvestigationResultStates.find((state) =>
          isPreviousState(state, process),
        )) && (
        <TenureInvestigationRecommendationBox
          processConfig={processConfig}
          process={process}
        />
      )}

      {(hoApprovalPassed.state === process.currentState.state ||
        isPreviousState(hoApprovalPassed.state, process)) && (
        <StatusBox
          variant="success"
          title={locale.views.hoReviewView.hoOutcome(
            locale.views.hoReviewModal[process.processName],
            "approved",
          )}
        >
          {hoApprovalPassedState?.processData.formData.reason && (
            <Text>{hoApprovalPassedState.processData.formData.reason}</Text>
          )}
        </StatusBox>
      )}

      {(hoApprovalFailed.state === process.currentState.state ||
        isPreviousState(hoApprovalFailed.state, process)) && (
        <StatusBox
          variant="warning"
          title={locale.views.hoReviewView.hoOutcome(
            locale.views.hoReviewModal[process.processName],
            "declined",
          )}
        >
          {hoApprovalFailedState?.processData.formData.reason && (
            <Text>{hoApprovalFailedState.processData.formData.reason}</Text>
          )}
        </StatusBox>
      )}

      {(nameUpdated.state === process.currentState.state ||
        isPreviousState(nameUpdated.state, process)) && (
        <StatusBox
          variant="success"
          title={locale.views.tenureInvestigation.tenancySigned}
        >
          <div
            style={{ marginTop: 15 }}
            className="govuk-link lbh-link lbh-link--no-visited-state"
          >
            <Link
              as={RouterLink}
              to={`/tenure/${
                (process.relatedEntities as any[])?.find(
                  (entity) =>
                    entity.targetType === "tenure" && entity.subType === "newTenure",
                )?.id
              }`}
              variant="link"
            >
              {locale.views.tenureInvestigation.viewNewTenure}
            </Link>
          </div>
        </StatusBox>
      )}
    </>
  );
};
