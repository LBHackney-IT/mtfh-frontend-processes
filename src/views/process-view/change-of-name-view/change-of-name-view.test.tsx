import React from "react";

import {
  getPersonV1,
  getProcessV1,
  mockPersonV1,
  mockProcessV1,
  render,
  server,
} from "@hackney/mtfh-test-utils";
import { screen, waitForElementToBeRemoved, within } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { locale } from "../../../services";
import { ProcessLayout } from "../process-layout";
import { ChangeOfNameSideBar, ChangeOfNameView } from "./change-of-name-view";

const options = {
  url: "/processes/changeofname/e63e68c7-84b0-3a48-b450-896e2c3d7735",
  path: "/processes/:processName/:processId",
};

const setCancel = jest.fn();
const setCloseProcessDialogOpen = jest.fn();

describe("changeofname/change-of-name-view", () => {
  beforeEach(() => {
    jest.resetModules();
  });

  test("it renders ChangeOfNameView correctly on state EnterNewName", async () => {
    server.use(getPersonV1(mockPersonV1));
    const { container } = render(
      <ChangeOfNameView
        process={{
          ...mockProcessV1,
          currentState: { ...mockProcessV1.currentState, state: "EnterNewName" },
        }}
        mutate={() => {}}
        optional={{
          closeProcessReasonFinal: "",
          submitted: false,
          setSubmitted: () => {},
          closeCase: false,
          setCloseCase: () => {},
        }}
      />,
      options,
    );
    await expect(
      screen.findByText(`${mockPersonV1.firstName} ${mockPersonV1.surname}`),
    ).resolves.toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test("it renders error correctly on state UnknownState", async () => {
    server.use(getPersonV1());
    render(
      <ChangeOfNameView
        process={{
          ...mockProcessV1,
          currentState: { ...mockProcessV1.currentState, state: "UnknownState" },
        }}
        mutate={() => {}}
        optional={{
          closeProcessReasonFinal: "",
          submitted: false,
          setSubmitted: () => {},
          closeCase: false,
          setCloseCase: () => {},
        }}
      />,
      options,
    );
    await expect(
      screen.findByText(locale.errors.unableToFindState),
    ).resolves.toBeInTheDocument();
  });

  test("it renders error if person cannot be fetched", async () => {
    server.use(getPersonV1({}, 500));
    render(
      <ChangeOfNameView
        process={{
          ...mockProcessV1,
          currentState: { ...mockProcessV1.currentState, state: "UnknownState" },
        }}
        mutate={() => {}}
        optional={{
          closeProcessReasonFinal: "",
          submitted: false,
          setSubmitted: () => {},
          closeCase: false,
          setCloseCase: () => {},
        }}
      />,
      options,
    );
    await expect(
      screen.findByText(locale.errors.unableToFetchRecord),
    ).resolves.toBeInTheDocument();
  });

  test("it renders ChangeOfNameSideBar correctly", async () => {
    const { container } = render(
      <ChangeOfNameSideBar
        process={{
          ...mockProcessV1,
          currentState: { ...mockProcessV1.currentState, state: "UnknownState" },
        }}
        submitted={false}
        closeCase={false}
        setCloseProcessDialogOpen={setCloseProcessDialogOpen}
        setCancel={setCancel}
      />,
      options,
    );
    await userEvent.click(
      screen.getByText(locale.views.changeofname.actions.cancelProcess),
    );
    expect(setCancel.mock.calls[0][0]).toBe(true);
    expect(container).toMatchSnapshot();
  });

  test("it renders ChangeOfNameSideBar correctly for NameSubmitted state", async () => {
    render(
      <ChangeOfNameSideBar
        process={{
          ...mockProcessV1,
          currentState: { ...mockProcessV1.currentState, state: "NameSubmitted" },
        }}
        submitted={false}
        closeCase={false}
        setCloseProcessDialogOpen={setCloseProcessDialogOpen}
        setCancel={setCancel}
      />,
      options,
    );

    const stepper = await screen.findByTestId("mtfh-stepper-change-of-name");
    const steps = within(stepper).getAllByRole("listitem");
    expect(steps[0].className).not.toContain("active");
    expect(steps[1].className).toContain("active");
  });

  test("it renders ChangeOfName correctly for DocumentsRequestedDes state", async () => {
    server.use(
      getProcessV1({
        ...mockProcessV1,
        currentState: { ...mockProcessV1.currentState, state: "DocumentsRequestedDes" },
      }),
    );
    render(<ProcessLayout />, options);

    const stepper = await screen.findByTestId("mtfh-stepper-change-of-name");
    const steps = within(stepper).getAllByRole("listitem");
    expect(steps[1].className).not.toContain("active");
    expect(steps[2].className).toContain("active");
    await expect(
      screen.findByTestId("changeofname-ReviewDocuments"),
    ).resolves.toBeInTheDocument();
  });

  test("it checks close case button", async () => {
    const setCancel = jest.fn();
    const setCloseProcessDialogOpen = jest.fn();
    server.use(getPersonV1());
    render(
      <ChangeOfNameView
        process={{
          ...mockProcessV1,
          currentState: { ...mockProcessV1.currentState, state: "DocumentsRequestedDes" },
        }}
        mutate={() => {}}
        optional={{
          closeProcessReasonFinal: "",
          submitted: false,
          setSubmitted: () => {},
          closeCase: false,
          setCloseCase: () => {},
          setCancel,
          setCloseProcessDialogOpen,
        }}
      />,
      options,
    );
    await waitForElementToBeRemoved(screen.queryAllByText(/Loading/));
    await userEvent.click(screen.getByText(locale.closeCase));
    expect(setCancel.mock.calls[0][0]).toBe(false);
    expect(setCloseProcessDialogOpen.mock.calls[0][0]).toBe(true);
  });

  test("it renders ChangeOfName correctly for DocumentChecksPassed state", async () => {
    server.use(
      getProcessV1({
        ...mockProcessV1,
        currentState: { ...mockProcessV1.currentState, state: "DocumentChecksPassed" },
      }),
    );
    render(<ProcessLayout />, options);

    const stepper = await screen.findByTestId("mtfh-stepper-change-of-name");
    const steps = within(stepper).getAllByRole("listitem");
    expect(steps[2].className).not.toContain("active");
    expect(steps[3].className).toContain("active");
    await expect(
      screen.findByText(locale.views.submitCase.tenureInvestigation),
    ).resolves.toBeInTheDocument();
  });

  test("it renders ChangeOfName sidebar correctly for ApplicationSubmitted state, submitted=true", async () => {
    render(
      <ChangeOfNameSideBar
        process={{
          ...mockProcessV1,
          currentState: { ...mockProcessV1.currentState, state: "ApplicationSubmitted" },
        }}
        submitted
        closeCase={false}
        setCloseProcessDialogOpen={setCloseProcessDialogOpen}
        setCancel={setCancel}
      />,
      options,
    );

    const stepper = await screen.findByTestId("mtfh-stepper-change-of-name");
    const steps = within(stepper).getAllByRole("listitem");
    expect(steps[3].className).not.toContain("active");
    expect(steps[4].className).toContain("active");
  });

  test("it renders ChangeOfName sidebar correctly for ApplicationSubmitted state, submitted false", async () => {
    render(
      <ChangeOfNameSideBar
        process={{
          ...mockProcessV1,
          currentState: { ...mockProcessV1.currentState, state: "ApplicationSubmitted" },
        }}
        submitted={false}
        closeCase={false}
        setCloseProcessDialogOpen={setCloseProcessDialogOpen}
        setCancel={setCancel}
      />,
      options,
    );

    const stepper = await screen.findByTestId("mtfh-stepper-change-of-name");
    const steps = within(stepper).getAllByRole("listitem");
    expect(steps[1].className).not.toContain("active");
    expect(steps[0].className).toContain("active");
  });

  test("it renders ChangeOfName correctly for ApplicationSubmitted state", async () => {
    server.use(
      getProcessV1({
        ...mockProcessV1,
        currentState: { ...mockProcessV1.currentState, state: "ApplicationSubmitted" },
      }),
    );
    render(<ProcessLayout />, options);

    await expect(
      screen.findByTestId("changeofname-tenure-investigation"),
    ).resolves.toBeInTheDocument();
  });
});
