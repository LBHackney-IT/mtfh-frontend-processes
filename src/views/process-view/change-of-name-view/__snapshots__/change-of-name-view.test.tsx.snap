// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changeofname/change-of-name-view it renders ChangeOfNameSideBar correctly 1`] = `
<div>
  <div
    data-testid="mtfh-stepper-change-of-name"
  >
    <div
      class="mtfh-stepper mtfh-stepper--large mtfh-stepper--active"
    >
      <ol
        class="mtfh-stepper"
      >
        <li
          class="mtfh-stepper__step mtfh-stepper__step--active"
        >
          <div
            class="mtfh-stepper__header"
          >
            <h4
              class="lbh-heading-h4 mtfh-stepper__title"
            >
              <span
                class="mtfh-stepper__circle mtfh-stepper__circle--number"
              >
                <span
                  class="mtfh-stepper__circle-inner"
                >
                  <span
                    class="mtfh-stepper__circle-background"
                  >
                    <span
                      class="govuk-visually-hidden"
                    >
                      Step
                    </span>
                    1
                  </span>
                </span>
              </span>
              <span>
                <button
                  class="mtfh-stepper__button mtfh-stepper__button--title"
                >
                  <span>
                    Tenant's new name
                  </span>
                </button>
              </span>
            </h4>
          </div>
        </li>
        <li
          class="mtfh-stepper__step"
        >
          <div
            class="mtfh-stepper__header"
          >
            <h4
              class="lbh-heading-h4 mtfh-stepper__title"
            >
              <span
                class="mtfh-stepper__circle mtfh-stepper__circle--number"
              >
                <span
                  class="mtfh-stepper__circle-inner"
                >
                  <span
                    class="mtfh-stepper__circle-background"
                  >
                    <span
                      class="govuk-visually-hidden"
                    >
                      Step
                    </span>
                    2
                  </span>
                </span>
              </span>
              <span>
                <button
                  class="mtfh-stepper__button mtfh-stepper__button--title"
                >
                  <span>
                    Request Documents
                  </span>
                </button>
              </span>
            </h4>
          </div>
        </li>
        <li
          class="mtfh-stepper__step"
        >
          <div
            class="mtfh-stepper__header"
          >
            <h4
              class="lbh-heading-h4 mtfh-stepper__title"
            >
              <span
                class="mtfh-stepper__circle mtfh-stepper__circle--number"
              >
                <span
                  class="mtfh-stepper__circle-inner"
                >
                  <span
                    class="mtfh-stepper__circle-background"
                  >
                    <span
                      class="govuk-visually-hidden"
                    >
                      Step
                    </span>
                    3
                  </span>
                </span>
              </span>
              <span>
                <button
                  class="mtfh-stepper__button mtfh-stepper__button--title"
                >
                  <span>
                    Review Documents
                  </span>
                </button>
              </span>
            </h4>
          </div>
        </li>
        <li
          class="mtfh-stepper__step"
        >
          <div
            class="mtfh-stepper__header"
          >
            <h4
              class="lbh-heading-h4 mtfh-stepper__title"
            >
              <span
                class="mtfh-stepper__circle mtfh-stepper__circle--number"
              >
                <span
                  class="mtfh-stepper__circle-inner"
                >
                  <span
                    class="mtfh-stepper__circle-background"
                  >
                    <span
                      class="govuk-visually-hidden"
                    >
                      Step
                    </span>
                    4
                  </span>
                </span>
              </span>
              <span>
                <button
                  class="mtfh-stepper__button mtfh-stepper__button--title"
                >
                  <span>
                    Submit case
                  </span>
                </button>
              </span>
            </h4>
          </div>
        </li>
        <li
          class="mtfh-stepper__step"
        >
          <div
            class="mtfh-stepper__header"
          >
            <h4
              class="lbh-heading-h4 mtfh-stepper__title"
            >
              <span
                class="mtfh-stepper__circle mtfh-stepper__circle--number"
              >
                <span
                  class="mtfh-stepper__circle-inner"
                >
                  <span
                    class="mtfh-stepper__circle-background"
                  >
                    <span
                      class="govuk-visually-hidden"
                    >
                      Step
                    </span>
                    5
                  </span>
                </span>
              </span>
              <span>
                <button
                  class="mtfh-stepper__button mtfh-stepper__button--title"
                >
                  <span>
                    Finish
                  </span>
                </button>
              </span>
            </h4>
          </div>
        </li>
      </ol>
    </div>
  </div>
  <button
    class="govuk-button lbh-button govuk-button--secondary lbh-button--secondary"
    type="button"
  >
    Reassign case
  </button>
  <button
    class="govuk-button lbh-button govuk-button--secondary lbh-button--secondary"
    type="button"
  >
    Cancel process
  </button>
  <a
    class="govuk-button lbh-button govuk-button--secondary lbh-button--secondary"
    href="/activities/process/ab/undefined"
  >
    Case activity history
  </a>
</div>
`;

exports[`changeofname/change-of-name-view it renders ChangeOfNameView correctly on state EnterNewName 1`] = `
<div>
  <h2
    class="lbh-heading-h2"
  >
    <div>
      <a
        class="govuk-link lbh-link lbh-link--no-visited-state"
        href="/person/6fbe024f-2316-4265-a6e8-d65a837e308a"
        target="_blank"
      >
        Haylee
         
        Macejkovic
      </a>
    </div>
  </h2>
  <div
    data-testid="changeofname-EnterNewName"
  >
    <h3
      class="lbh-heading-h3"
    >
      Enter tenant's new name
    </h3>
    <form
      action="#"
      class="mtfh-person-form"
      id="person-form"
    >
      <div
        class="govuk-form-group lbh-form-group"
        id="person-form-new-tenant-name"
      >
        <label
          class="govuk-label lbh-label"
          for="person-form-new-tenant-name-field"
        >
          
        </label>
        <div
          class="govuk-form-group lbh-form-group"
          id="person-form-title"
        >
          <label
            class="govuk-label lbh-label"
            for="person-form-title-field"
          >
            Title
            <sup
              aria-hidden="true"
            >
              *
            </sup>
          </label>
          <select
            class="govuk-select lbh-select mtfh-person-form__input--contained"
            id="person-form-title-field"
            name="title"
            required=""
          >
            <option
              value=""
            >
              Select Title
            </option>
            <option
              value="Mr"
            >
              Mr
            </option>
            <option
              value="Mrs"
            >
              Mrs
            </option>
            <option
              value="Miss"
            >
              Miss
            </option>
            <option
              value="Ms"
            >
              Ms
            </option>
            <option
              value="Dr"
            >
              Dr
            </option>
            <option
              value="Master"
            >
              Master
            </option>
            <option
              value="Rabbi"
            >
              Rabbi
            </option>
            <option
              value="Reverend"
            >
              Reverend
            </option>
            <option
              value="Other"
            >
              Other
            </option>
          </select>
        </div>
        <div
          class="govuk-form-group lbh-form-group"
          id="person-form-firstName"
        >
          <label
            class="govuk-label lbh-label"
            for="person-form-firstName-field"
          >
            First name
            <sup
              aria-hidden="true"
            >
              *
            </sup>
          </label>
          <input
            class="govuk-input lbh-input"
            id="person-form-firstName-field"
            name="firstName"
            placeholder="Enter first name"
            required=""
            value=""
          />
        </div>
        <div
          class="govuk-form-group lbh-form-group"
          id="person-form-middleName"
        >
          <label
            class="govuk-label lbh-label"
            for="person-form-middleName-field"
          >
            Middle name
            
          </label>
          <input
            class="govuk-input lbh-input"
            id="person-form-middleName-field"
            name="middleName"
            placeholder="Enter Middle name"
            value=""
          />
        </div>
        <div
          class="govuk-form-group lbh-form-group"
          id="person-form-surname"
        >
          <label
            class="govuk-label lbh-label"
            for="person-form-surname-field"
          >
            Last name
            <sup
              aria-hidden="true"
            >
              *
            </sup>
          </label>
          <input
            class="govuk-input lbh-input"
            id="person-form-surname-field"
            name="surname"
            placeholder="Enter last name"
            required=""
            value=""
          />
        </div>
      </div>
      <div
        class="start-process__actions"
      >
        <button
          class="govuk-button lbh-button"
          disabled=""
          type="submit"
        >
          Next
        </button>
      </div>
    </form>
  </div>
</div>
`;
